### DISCUSSION TITLE
Using closure for ViewColumn data parameter

### DISCUSSION THREAD

**Discussion by myregistration** [filamentphp/filament#14008] [Help]
### Package

Table builder

### Package Version

v3

### How can we help you?

When using `Tables\Columns\ViewColumn::make()` I want to pass particular data to the view based on the `$record` relationships, but it won't allow me to use a closure in the` view()` or `viewData()` methods which would allow me to access $record.

**Comment by myregistration** [comment]
Looks like I can use the following as an alternative.  Unfortunately, you have to pass a valid field name to the `TextColumn::make()` method or it won't process.

```
Tables\Columns\TextColumn::make('full_name')
                    ->label('User')
                    ->formatStateUsing(fn (string $state, $record): View =>  view('filament.tables.columns.user', ['user' => $record]),
```

**Comment by dissto** [comment]
Don't you have access to `$getRecord()` by default when using`->view(...)`?  🤔

**Reply by myregistration** [reply]
Yes, but I am trying to create reusable blade components and sometimes the record is relative to a relationship, not the actual record.

**Comment by jirapetr** [comment]
I have the same problem :

`ViewColumn::make('id')->view('components.url-button', [
 'href' => route('filament.admin.resources.documents.create', [
 'employee_id'           => $record->id,
                        ]),
])
```

**Comment by peter-mw** [comment]
any news on this ?

**Comment by 00bx** [comment]
u dont have to use any field for the text column approach, u can use ->state(true) this gives it pre state before modifies it to the view state, this way u can pass any data u want, as an example:  
Tables\Columns\TextColumn::make('test')
                ->formatStateUsing(function ($record){
                    return view('example',compact('record'));
                })
                ->state(true),

**Reply by jirapetr** [reply]
and this?
Tables\Columns\TextColumn::make('test')
    ->formatStateUsing(function ($record) {
        return view('example', compact('record'))->render();
    });

**Reply by 00bx** [reply]
based on the tests, if the column doesnt exist the state will set to null always, so u have to give it a pre state then modify state, its like saying.. theres no state what do u want me to modify.
